core/src/test/java/com/radixdlt/consensus/safety/SafetyStoreMigrationTest.java

```diff
@@ -0,0 +1,170 @@
+package com.radixdlt.consensus.safety;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.inject.*;
+import com.google.inject.Module;
+import com.radixdlt.consensus.bft.BFTValidatorId;
+import com.radixdlt.environment.NodeRustEnvironment;
+import com.radixdlt.monitoring.Metrics;
+import com.radixdlt.monitoring.MetricsInitializer;
+import com.radixdlt.rev2.NodeRustEnvironmentBuilder;
+import com.radixdlt.rev2.modules.NodePersistenceModule;
+import com.radixdlt.rev2.modules.NodePersistenceModule.AddressBookPolicy;
+import com.radixdlt.serialization.Serialization;
+import com.radixdlt.serialization.core.ClasspathScanningSerializationPolicy;
+import com.radixdlt.serialization.core.ClasspathScanningSerializerIds;
+import com.radixdlt.store.BerkeleyDbDefaults;
+import com.radixdlt.store.NodeStorageLocation;
+import com.radixdlt.utils.properties.RuntimeProperties;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.commons.cli.ParseException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class SafetyStoreMigrationTest {
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private Module createCommonModule() {
+    return new AbstractModule() {
+      @Provides
+      @Singleton
+      @NodeStorageLocation
+      String nodeStorageLocation() {
+        return folder.getRoot().getPath();
+      }
+
+      @Provides
+      @Singleton
+      NodeRustEnvironment nodeRustEnvironment() throws IOException {
+        return NodeRustEnvironmentBuilder.createNodeRustEnvironment(folder.newFolder().getPath());
+      }
+
+      @Provides
+      @Singleton
+      Metrics metrics() {
+        return new MetricsInitializer().initialize();
+      }
+
+      @Provides
+      @Singleton
+      Serialization serialization() {
+        return Serialization.create(
+            ClasspathScanningSerializerIds.create(), ClasspathScanningSerializationPolicy.create());
+      }
+
+      @Provides
+      @Singleton
+      RuntimeProperties properties() throws ParseException {
+        return RuntimeProperties.defaultWithOverrides(Map.of());
+      }
+    };
+  }
+
+  private Module createBerkeleyModule() {
+    return new AbstractModule() {
+      @Provides
+      @Singleton
+      BerkeleySafetyStateStore safetyStore(
+          RuntimeProperties properties,
+          Serialization serialization,
+          Metrics metrics,
+          @NodeStorageLocation String nodeStorageLocation) {
+        return new BerkeleySafetyStateStore(
+            serialization,
+            metrics,
+            nodeStorageLocation,
+            BerkeleyDbDefaults.createDefaultEnvConfigFromProperties(properties));
+      }
+    };
+  }
+
+  @Test
+  public void address_book_migrates_successfully() {
+    var injector =
+        Guice.createInjector(
+            createCommonModule(),
+            new NodePersistenceModule(AddressBookPolicy.DISABLED),
```

Comment:
The address book persistence state should be irrelevant to this test - something feels off here

Reply:
Reworked.